lib/browser/navigation.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+var util = require('util');
+var Url = require('url');
+
+var debug = require('debug')('testium-driver-sync:navigation');
+var _ = require('lodash');
+var Asserter = require('wd').Asserter;
+
+function compare(expected, actual) {
+  debug('Compare: %s === %s', expected, actual);
+  if (typeof expected === 'string') {
+    return expected === actual;
+  } else if (_.isRegExp(expected)) {
+    return expected.test(actual);
+  }
+  throw new Error('Invalid assertion: ' + expected);
+}
+
+function compareAllowNull(expected, actual) {
+  if (expected === null || expected === undefined) {
+    debug('Skipping: %s === %s', expected, actual);
+    return true;
+  }
+  return compare(expected, actual);
+}
+
+function propertyAsserter(property, expected) {
+  var getter = 'get' + property[0].toUpperCase() + property.slice(1);
+  return new Asserter(function checkProperty(browser) {
+    return browser[getter]().then(function compareWith(actual) {
+      if (!compare(expected, actual)) {
+        var err = new Error(util.format(
+          'Timed out waiting for %s %s. Last value was: %j',
+          property, expected, actual));
+        err.retriable = true;
+        throw err;
+      }
+    });
+  });
+}
+
+var BASIC_PROPERTIES = [
+  'protocol', 'host', 'pathname', 'hash',
+];
+function urlAsserter(url, query) {
```

Comment:
This should be functionally equivalent to https://github.com/testiumjs/testium-driver-sync/blob/master/lib/browser/makeUrlRegExp.js.